File: AED/Laravel/Restaurante/app/Models/Reserva.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use \Datetime;

class Reserva extends Model
{
    //use HasFactory;
    private int $id_reserva;
    private int $telefono;
    private Datetime $fecha_hora;
    private int $duracion;

    public function __construct(int $id_reserva = 0, int $telefono=0, Datetime $fecha_hora = new Datetime(), int $duracion=0)
    {
        $this->id_reserva = $id_reserva;
        $this->telefono = $telefono;
        $this->fecha_hora = $fecha_hora;
        $this->duracion = $duracion;
    }




    /**
     * Get the value of id_reserva
     */ 
    public function getId_reserva()
    {
        return $this->id_reserva;
    }

    /**
     * Set the value of id_reserva
     *
     * @return  self
     */ 
    public function setId_reserva($id_reserva)
    {
        $this->id_reserva = $id_reserva;

        return $this;
    }

    /**
     * Get the value of telefono
     */ 
    public function getTelefono()
    {
        return $this->telefono;
    }

    /**
     * Set the value of telefono
     *
     * @return  self
     */ 
    public function setTelefono($telefono)
    {
        $this->telefono = $telefono;

        return $this;
    }

    /**
     * Get the value of fecha_hora
     */ 
    public function getFecha_hora()
    {
        return $this->fecha_hora;
    }

    /**
     * Set the value of fecha_hora
     *
     * @return  self
     */ 
    public function setFecha_hora($fecha_hora)
    {
        $this->fecha_hora = $fecha_hora;

        return $this;
    }

        /**
         * Get the value of duracion
         */ 
        public function getDuracion()
        {
                return $this->duracion;
        }

        /**
         * Set the value of duracion
         *
         * @return  self
         */ 
        public function setDuracion($duracion)
        {
                $this->duracion = $duracion;

                return $this;
        }
}

File: AED/JPA/reservas/src/main/java/es/iespuertodelacruz/saul/entities/Reserva.java
package es.iespuertodelacruz.saul.entities;

import java.io.Serializable;
import javax.persistence.*;
import java.math.BigInteger;


/**
 * The persistent class for the reservas database table.
 * 
 */
@Entity
@Table(name="reservas")
@NamedQuery(name="Reserva.findAll", query="SELECT r FROM Reserva r")
@NamedQuery(name="Reserva.findByEstado", query="SELECT r FROM Reserva r WHERE r.estado = :estado")
@NamedQuery(name="Reserva.findByTelefono", query="SELECT r FROM Reserva r WHERE r.usuario.telefono = :telefono")
//@NamedQuery(name= "Alumno.findAllRel", query = "SELECT a FROM Alumno a JOIN FETCH a.matriculas")
public class Reserva implements Serializable {
	@Override
	public String toString() {
		return "Reserva [idReserva=" + idReserva + ", duracion=" + duracion + ", estado=" + estado + ", fechaHora="
				+ fechaHora + ", mesa=" + mesa + ", usuario=" + usuario + "]";
	}

	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="id_reserva")
	private int idReserva;

	private int duracion;

	private String estado;

	@Column(name="fecha_hora")
	private BigInteger fechaHora;

	//bi-directional many-to-one association to Mesa
	@ManyToOne
	@JoinColumn(name="num_mesa")
	private Mesa mesa;

	//bi-directional many-to-one association to Usuario
	@ManyToOne
	@JoinColumn(name="telefono")
	private Usuario usuario;

	public Reserva() {
	}

	public int getIdReserva() {
		return this.idReserva;
	}

	public void setIdReserva(int idReserva) {
		this.idReserva = idReserva;
	}

	public int getDuracion() {
		return this.duracion;
	}

	public void setDuracion(int duracion) {
		this.duracion = duracion;
	}

	public String getEstado() {
		return this.estado;
	}

	public void setEstado(String estado) {
		this.estado = estado;
	}

	public BigInteger getFechaHora() {
		return this.fechaHora;
	}

	public void setFechaHora(BigInteger fechaHora) {
		this.fechaHora = fechaHora;
	}

	public Mesa getMesa() {
		return this.mesa;
	}

	public void setMesa(Mesa mesa) {
		this.mesa = mesa;
	}

	public Usuario getUsuario() {
		return this.usuario;
	}

	public void setUsuario(Usuario usuario) {
		this.usuario = usuario;
	}

}